From 2b10fb586e5b812c0e85af34a1d1339503ede557 Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@balena.io>
Date: Mon, 7 Apr 2025 05:10:33 +0000
Subject: [PATCH] overlays: Add waveshare-sim7600 dtbo

This adds the necessary dtbo for GPIO
setup of the Waveshare SIM7600 modem.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Florin Sarbu <florin@balena.io>
---
 arch/arm/boot/dts/overlays/Makefile           |  1 +
 .../overlays/waveshare-sim7600-overlay.dts    | 20 +++++++++++++++++++
 2 files changed, 21 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/waveshare-sim7600-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index ba3f7d0f3ab5..cec30a3cb4df 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -355,6 +355,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	watterott-display.dtbo \
 	waveshare-can-fd-hat-mode-a.dtbo \
 	waveshare-can-fd-hat-mode-b.dtbo \
+	waveshare-sim7600.dtbo \
 	wifimac.dtbo \
 	wittypi.dtbo \
 	wm8960-soundcard.dtbo \
diff --git a/arch/arm/boot/dts/overlays/waveshare-sim7600-overlay.dts b/arch/arm/boot/dts/overlays/waveshare-sim7600-overlay.dts
new file mode 100644
index 000000000000..7295ecec1220
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/waveshare-sim7600-overlay.dts
@@ -0,0 +1,20 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+    compatible = "brcm,bcm2708,bcm2710,bcm2711";
+
+    fragment@0 {
+        target = <&gpio>;
+        __overlay__ {
+            pinctrl-names = "default";
+            pinctrl-0 = <&modem_pins>;
+
+            modem_pins: modem_pins {
+                brcm,pins = <4 6>;     /* GPIO pins */
+                brcm,function = <1 1>; /* Output */
+                brcm,pull = <2 2>; /* pull high */
+            };
+        };
+    };
+};
-- 
2.34.1

